test/ComponentInstallerTest.php

```diff
@@ -2031,15 +1654,65 @@ private function createConfigFile($name, $contents): void
     }
 
     /**
-     * @param PackageEvent|ObjectProphecy $event
-     * @psalm-param PackageEvent&ObjectProphecy $event
+     * @param PackageEvent|MockObject $event
+     * @psalm-param PackageEvent&MockObject $event
      */
     private function prepareEventForPackageProviderDetection($event, string $packageName): void
     {
         if (method_exists(PackageEvent::class, 'getPool')) {
-            $pool = $this->prophesize(Pool::class);
-            $pool->whatProvides($packageName)->willReturn([]);
-            $event->getPool()->willReturn($pool->reveal());
+            $pool = $this->createMock(Pool::class);
+            $pool->method('whatProvides')->with($packageName)->willReturn([]);
+            $event->method('getPool')->willReturn($pool);
         }
     }
+
+    /**
+     * @param array<int,string> $informations
+     */
+    private function createOutputAssertions(array $informations): void
+    {
+        $consecutiveArguments = [];
+
+        foreach ($informations as $information) {
+            $consecutiveArguments[] = [
+                self::callback(static function (string $argument) use ($information): bool {
+                    return preg_match(
+                        sprintf('/%s/', preg_quote($argument, '/')),
+                        $information
+                    ) !== false;
+                })
+            ];
+        }
+
+        $this->io
+            ->expects(self::exactly(count($consecutiveArguments)))
+            ->method('write')
+            ->withConsecutive(...$consecutiveArguments);
+    }
+
+    /**
+     * @param array<int,AbstractQuestionAssertion> $questionsAssertions
+     */
+    private function createInputAssertions(array $questionsAssertions): void
+    {
+        $consecutiveReturnValues = $consecutiveArguments = [];
+        foreach ($questionsAssertions as $questionAssertion) {
+            /** @psalm-suppress MissingClosureParamType */
+            $consecutiveArguments[] = [
+                self::callback($questionAssertion->assertion())
+            ];
+            $consecutiveReturnValues[] = $questionAssertion->expectedAnswer;
+
+            if ($questionAssertion instanceof RememberedAnswerQuestionAssertion) {
+                $consecutiveArguments[] = [self::callback($questionAssertion->rememberAnswerAssertion())];
+                $consecutiveReturnValues[] = $questionAssertion->remember ? 'y' : 'n';
+            }
+        }
+
+        $this->io
+            ->expects(self::exactly(count($consecutiveArguments)))
+            ->method('ask')
+            ->withConsecutive(...$consecutiveArguments)
+            ->willReturnOnConsecutiveCalls(...$consecutiveReturnValues);
+    }
```

Comment:
Please move these to near the top of the classfile. Generally speaking, we like to keep the setup/teardown, assertions, and any utility methods at the top, to make them easily discoverable, as new tests are generally _appended_ to the test classes.